packages/astro/src/runtime/server/render/astro/render.ts

```diff
@@ -165,3 +166,101 @@ async function bufferHeadContent(result: SSRResult) {
 		}
 	}
 }
+
+export async function renderToAsyncIterable(
+	result: SSRResult,
+	componentFactory: AstroComponentFactory,
+	props: any,
+	children: any,
+	isPage = false,
+	route?: RouteData
+): Promise<AsyncIterable<Uint8Array> | Response> {
+  const templateResult = await callComponentAsTemplateResultOrResponse(
+    result,
+    componentFactory,
+    props,
+    children,
+    route
+  );
+  if (templateResult instanceof Response)
+    return templateResult;
+  let renderedFirstPageChunk = false;
+  if (isPage) {
+    await bufferHeadContent(result);
+  }
+
+	let error: Error | null = null;
+  let next = promiseWithResolvers<void>();
+  const chunks: Uint8Array[] = []; // []Uint8Array
+
+  const iterator = {
+    async next() {
+      await next.promise;
+
+			if(error) {
+				throw error;
+			}
+
+      // Get the total length of all arrays.
+      let length = 0;
+      chunks.forEach(item => {
+        length += item.length;
+      });
```

Comment:
nit: let's use a `const length = chunks.map()` instead

Reply:
You mean reduce? Why though? it's equivalent code.

Reply:
Switched to a plain for loop, let me know what you think.

Reply:
Ah sorry I missed the `+=`. Yeah for loop was the other suggestion